### ISSUE TITLE
Feature request: Parallel shell commands

### ISSUE THREAD

**Issue by Zash** [buildbot/buildbot#4734]
In a scenario where one runs tests against a daemon it would be nice to catch logs from both the daemon and whatever testing tool. However this seems tricky to do without any way to run two commands in parallel.

**Comment by p12tic** [comment]
Sounds like this should not be a rare use case.

**Comment by nosuchip** [comment]
Faced same situation. @Zash how did you started daemon with Buildbot step?

**Comment by Zash** [comment]
Found out about `logfiles`, so it's possible to have the daemon log to a file and pick that up from the test tool, like this:

```python
test_factory.addSteps([
    # clone, build, etc.
    steps.ShellCommand(command=["./prosodyctl", "start" ]), # starts and daemonizes, logs to prosody.log
    steps.ShellCommand(command=["scansion"], logfiles={"prosody":"prosody.log"}),
    steps.ShellCommand(command=["./prosodyctl", "stop" ]),
])
```

[Prosody](https://prosody.im) is the daemon here and [scansion](https://matthewwild.co.uk/projects/scansion/) the test runner.

---

It would still be nice to have a way to run two foreground commands, capturing stdout from both, if nothing else for proper process control. If something goes wrong here I fear it may leave the daemonized process running, basically leaking it.

**Comment by Zash** [comment]
> If something goes wrong here I fear it may leave the daemonized process running, basically leaking it.

Something finally went wrong and we were leaking processes with each test run.

Any ideas for how one might implement this?
